File: week6/sentimental-readability/readability.py
# TODO
import cs50

frase = cs50.get_string("Text: ")

letras=0
sentencas=0
palavras=0
whitespace=' '
end_of_sentences= ['!', '.', '?']

letras = len([ele for ele in frase if ele.isalpha()])
palavras=len([ele for ele in frase if ele==whitespace])+1
sentencas = len([ele for ele in frase if ele in end_of_sentences])
# if sentencas==0:
#     sentencas=1
# print("Palavras: "+str(palavras))
# print("Letras: " + str(letras))
# print("Sentencas: " + str(sentencas))
le=float(letras)
pa=float(palavras)
se=float(sentencas)


L=le*100/pa
S=se*100/pa

# print("L: {}".format(L))
# print("S: {}".format(S))

G=0.0588 * L - 0.296 * S - 15.8
print(G)
if G<0:
    grade=G-0.5
else:
    grade=G+0.5

grade1=int(grade)
if (grade < 1):
        print("Before Grade 1\n");
elif grade1>16:
    print("Grade: 16+")
else:
    print("Grade: {}".format(grade1))

File: week2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>

int main(void)
{
    double words = 1;
    double letters = 0;
    double chars = 0;
    double sentences=0;
    //---------passar a quantidade de letras da string num vetor em char------
    string texto = get_string("Text: ");

    char vetor[strlen(texto)];

    int i;
    for (i = 0; i < strlen(texto); i++) {
        vetor[i] = texto[i];
    }

    for (i=0;i<strlen(texto);i++){
        if (vetor[i] == ' '){//eu havia colocado apenas vetor, porém ocorreu erro de segmentation fault, implemetado [i] para resolver.
            words++;
        }

        if (vetor[i] == '.' || vetor[i] == '!' || vetor[i] == '?'){
            sentences++;
        }

        if(vetor[i] >= 'a' && vetor[i] <= 'z'){
            letters++;
        }

        if(vetor[i] >= 'A' && vetor[i] <= 'Z'){
            letters++;
        }
    }


    printf("A qtd de palavras é : %f\n", words);
    printf("A qtd de letras é : %f\n", letters);
    printf("A qtd de sentenças é: %f\n", sentences);

    double letters_avg = letters / words * 100;
    double sentences_avg = sentences / words * 100;
    double temp_grade = 0.0588 * letters_avg - 0.296 * sentences_avg - 15.8;

    printf("L: %f\n", letters_avg);
    printf("S: %f\n", sentences_avg);
    printf("G: %f\n", temp_grade);

    int grade = (int)(temp_grade < 0 ? (temp_grade - 0.5) : (temp_grade + 0.5));

    if (grade < 1) {
        printf("Before Grade 1\n");
    } else if (grade > 16) {
        printf("Grade 16+\n");
    } else {
        printf("Grade %i\n", grade);
    }
    return 0;
}
